### PR TITLE
remove auth.currentUser from frontend endpoints

### LINKED ISSUE #734: Pass UIDs to Backend From Middleware Rather Than Part of the Request Body from the Frontend
**Describe the solution you'd like**
Backend endpoints should get the firebase user id from the middleware token verification logic rather than as part of the frontend request. This is to ensure that the uid used by any backend endpoint matches up with the bearer token passed in to the middleware (so that user A cannot make a request to the backend with the uid of user B, for example).

**Setup Instructions (what branch to work off of)**
Run the following commands
```
git checkout dev
git pull origin dev
git checkout -b nextjs
```
FYI: If you are not able to immediately run `git checkout next's`, make sure you commit your changes in the current branch or run `git stash` and then execute the above commands

### PR DESCRIPTION
**Pull request by archishmagoli**
# Remove auth.currentUser from frontend endpoints

**What user problem are we solving?**
`auth.currentUser` is being unnecessarily passed in the frontend endpoint queries.

**What solution does this PR provide?**
Remove the parameter in frontend endpoint files.

**Testing Methodology**
Used Postman to verify that requests to existing backend endpoints are still successful.

**Any other considerations**
N/A

### CODE REVIEW

**Review by karkir0003 — COMMENTED** [review]
conditionally approved. wait on @dwu359 to review?

**Review by farisdurrani — APPROVED** [review]

**Comment by farisdurrani** [comment]
Await karthik's or daniel's approval

**Review by karkir0003 — APPROVED** [review]

**Review by dwu359 — COMMENTED** [review]
The auth protected apis (at least the ones involving training & trainspaces) don't seem to work for me, because those apis require the user's uid, which is originally received from auth.currentUser. Is a way for the backend to get the uid without auth.currentUser already implemented (such as through the bearer token via middleware)?

**Comment by karkir0003** [comment]
> The auth protected apis (at least the ones involving training & trainspaces) don't seem to work for me, because those apis require the user's uid, which is originally received from auth.currentUser. Is a way for the backend to get the uid without auth.currentUser already implemented (such as through the bearer token via middleware)?

@dwu359 I believe the backend does populate the `"user"` field. See this [line](https://github.com/DSGT-DLP/Deep-Learning-Playground/blob/nextjs/backend/middleware.py#L43)

`authenticate` function: [here](https://github.com/DSGT-DLP/Deep-Learning-Playground/blob/nextjs/backend/firebase_helpers/authenticate.py#L11)

**Comment by karkir0003** [comment]
> The auth protected apis (at least the ones involving training & trainspaces) don't seem to work for me, because those apis require the user's uid, which is originally received from auth.currentUser. Is a way for the backend to get the uid without auth.currentUser already implemented (such as through the bearer token via middleware)?

how did you test the train endpoints? @dwu359

**Comment by dwu359** [comment]
> > The auth protected apis (at least the ones involving training & trainspaces) don't seem to work for me, because those apis require the user's uid, which is originally received from auth.currentUser. Is a way for the backend to get the uid without auth.currentUser already implemented (such as through the bearer token via middleware)?
> 
> how did you test the train endpoints? @dwu359

I tested them by running the website itself in dev mode, and the apis returned a 400 error for me. The getUserDatasetFilesData endpoint returns a 400 error saying that the request_data["user"]["uid"] cannot be found, and the s3_endpoint returns a 400 error saying that for json.loads(request.args) the JSON object must be str, bytes or bytearray, not ImmutableMultiDict

**Comment by archishmagoli** [comment]
Thanks for the feedback, @dwu359. I'm currently looking into this, but if we're already authenticating through the middleware, then can we get the UID through `environ["user"]` (which is the `"user"` field Karthik mentioned) instead of `auth.currentUser` in the frontend? If we're able to access the UID this way then I think we can fix the endpoint errors by just shifting the source away from `request.args`. 

(Otherwise we can use the bearer approach I guess.)

**Comment by archishmagoli** [comment]
Update: looks like we _can_ access the `"user"` field from `request.environ["user"]["uid"]` in the backend. Now I'm getting a 200 on the `getUserDatasetFilesData` endpoint. I added a commit with the quick fix.

However, I'm getting another error through `img_run` for one of the `train_endpoints` - full error message is pasted below. Not sure if this a previously existing issue though: 

`{"success": false, "message": "Traceback (most recent call last):
File \"C:\\Users\\archi\\Deep-Learning-Playground\\backend\\endpoints\\train_endpoints.py\", line 121, in img_run\  train_transform = request_data[\"train_transform\"]
KeyError: 'train_transform'"}`

**Comment by karkir0003** [comment]
> Update: looks like we _can_ access the `"user"` field from `request.environ["user"]["uid"]` in the backend. Now I'm getting a 200 on the `getUserDatasetFilesData` endpoint. I added a commit with the quick fix.
> 
> However, I'm getting another error through `img_run` for one of the `train_endpoints` - full error message is pasted below. Not sure if this a previously existing issue though:
> 
> `{"success": false, "message": "Traceback (most recent call last): File \"C:\\Users\\archi\\Deep-Learning-Playground\\backend\\endpoints\\train_endpoints.py\", line 121, in img_run\ train_transform = request_data[\"train_transform\"] KeyError: 'train_transform'"}`

so image train needs to have a train transform specified or provide "train_transform: []" @archishmagoli

File: backend/endpoints/trainspace_endpoints.py
Comment (karkir0003) [inline backend/endpoints/trainspace_endpoints.py]: is this commented out code needed @archishmagoli

**Comment by archishmagoli** [comment]
#

> > Update: looks like we _can_ access the `"user"` field from `request.environ["user"]["uid"]` in the backend. Now I'm getting a 200 on the `getUserDatasetFilesData` endpoint. I added a commit with the quick fix.
> > However, I'm getting another error through `img_run` for one of the `train_endpoints` - full error message is pasted below. Not sure if this a previously existing issue though:
> > `{"success": false, "message": "Traceback (most recent call last): File \"C:\\Users\\archi\\Deep-Learning-Playground\\backend\\endpoints\\train_endpoints.py\", line 121, in img_run\ train_transform = request_data[\"train_transform\"] KeyError: 'train_transform'"}`
> 
> so image train needs to have a train transform specified or provide "train_transform: []" @archishmagoli

Okay, so that's a user issue lol. We should be all good then.

File: backend/endpoints/trainspace_endpoints.py
Comment (archishmagoli) [reply backend/endpoints/trainspace_endpoints.py]: No, I can remove it if the fix works.

**Comment by karkir0003** [comment]
@archishmagoli can you try img-run endpoint with that insight and see if you can get a successful api response?

**Comment by karkir0003** [comment]
> Update: looks like we _can_ access the `"user"` field from `request.environ["user"]["uid"]` in the backend. Now I'm getting a 200 on the `getUserDatasetFilesData` endpoint. I added a commit with the quick fix.
> 
> However, I'm getting another error through `img_run` for one of the `train_endpoints` - full error message is pasted below. Not sure if this a previously existing issue though:
> 
> `{"success": false, "message": "Traceback (most recent call last): File \"C:\\Users\\archi\\Deep-Learning-Playground\\backend\\endpoints\\train_endpoints.py\", line 121, in img_run\ train_transform = request_data[\"train_transform\"] KeyError: 'train_transform'"}`

also, the middleware invokes `request.environ` to store the `['user']`  entry

**Review by dwu359 — APPROVED** [review]
Works now

**Comment by archishmagoli** [comment]
Awesome. I removed the comments as per @karkir0003 and expanded coverage of `request.environ` to one of the endpoints I missed. Will merge ASAP.

**Review by karkir0003 — COMMENTED** [review]
looks good to me. @dwu359 to confirm?

**Comment by dwu359** [comment]
Yep looks good to me

**Pull request merged**
